### PR TITLE
Improve the ergonomics of data loading in the front-end

### PR DESCRIPTION
**Pull request by raix**
### Summary & Motivation

Wrap the `openapi-fetch` client as an easy to use client specific to React and PlatformPlatform.

* Add form `Action` support matching the new React api's `useFormState`/`useActionState`
* Add a `useApi` hook for easy data loading, with build in state management
* Ensure the api client always return errors in the `ProblemDetails` format

#### React hook: `useApi` - Simple
The hook can be used to simply get the `data` from the server, but also provides `loading`, `success` state and problem detail info like `type` / `status` / `title` / `detail` and `errors` compatible with `React Aria Components` validation in forms.
```ts
  const { data } = useApi("/api/account-management/users", {
    params: {
      query: {
        PageSize: 1
      }
    }
  });
```

#### React hook: `useApi` - Debounced fetching

For use with debounced search etc. and threshold of min 3 chars before fetching the result:
```ts
  const [subdomain, setSubdomain] = useState("");
  const { data: isSubdomainFree } = useApi(
    "/api/account-management/account-registrations/is-subdomain-free",
    {
      params: {
        query: { Subdomain: subdomain }
      }
    },
    {
      autoFetch: subdomain.length > 3,
      debounceMs: 500
    }
  );
```

#### Use with `useFormState` / `useActionState`

The `action` is passed to the `Form` component and `errors` can be passed directly to `validationErrors`:

```ts
  const [{ success, errors, data, title, message }, action, isPending] = useFormState(
    api.action("/api/account-management/account-registrations/start"),
    { success: null }
  );

  // ...

    <Form
      action={action}
      validationErrors={errors}
      validationBehavior="aria"
    >
```

#### Plain fetch

The result will always be the data and the api client throws `ProblemDetailsError`
```ts
  const result = await api.get("/api/account-management/account-registrations/is-subdomain-free",
    {
      params: {
        query: { Subdomain: subdomain }
      }
    });
```

### Checklist

- [x] I have added a Label to the pull-request
- [x] I have added tests, and done manual regression tests
- [x] I have updated the documentation, if necessary

### CODE REVIEW

**Review by tjementum — CHANGES_REQUESTED** [review]
Let me take a look. Please update the pull-request description, so it includes all the details of what is changed and why. It's the only dokumentation we have :)

File: application/account-management/Application/AccountRegistrations/StartAccountRegistration.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public TenantId GetTenantId()
     }
 }
 
-public sealed record StartAccountRegistrationResponse(string AccountRegistrationId, int ValidForSeconds);
+public sealed record StartAccountRegistrationResponse(string AccountRegistrationId, string Email, int ValidForSeconds);
```
Comment (tjementum) [inline application/account-management/Application/AccountRegistrations/StartAccountRegistration.cs:23]: I don't see this is used in the frontend. It does not belong in the response, as the only thing we need for `CompleteAccountRegistration` is the `AccountRegistrationId`.

Am I missing something?

Comment (raix) [reply application/account-management/Application/AccountRegistrations/StartAccountRegistration.cs:23]: It makes the FE code much cleaner + we can trust that the data in this response

File: application/shared-kernel/ApiCore/SchemaProcessor/StronglyTypedDocumentProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            .Where(t => typeof(IStronglyTypedId).IsAssignableFrom(t))
+            .Select(t => new { t.Name, t.GetCustomAttribute<IdPrefixAttribute>()?.Prefix })
+            .ToList();
+
+        // Ensure the Swagger UI to correctly display strongly typed IDs as plain text instead of complex objects
+        foreach (var stronglyTypedIdName in stronglyTypedIdNames)
+        {
+            if (!context.Document.Components.Schemas.TryGetValue(stronglyTypedIdName.Name, out var schema)) continue;
+            schema.Type = JsonObjectType.String;
+            if (stronglyTypedIdName.Prefix != null)
```
Comment (tjementum) [inline application/shared-kernel/ApiCore/SchemaProcessor/StronglyTypedDocumentProcessor.cs:24]:
```suggestion
            if (stronglyTypedIdName.Prefix is not null)
```

**Review by tjementum — APPROVED** [review]
I don't have more comments, except for pull-request description. And I would always a bit cleaner commit history.

But overall, this is just awesome :)

**Pull request merged**